File: php/features.php
<?php
declare(strict_types=1);

// Cartoons SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CartoonsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CartoonsBaseFeature();
            case "test":
                return new CartoonsTestFeature();
            default:
                return new CartoonsBaseFeature();
        }
    }
}

File: py/features.py
# Cartoons SDK feature factory

from feature.base_feature import CartoonsBaseFeature
from feature.test_feature import CartoonsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CartoonsBaseFeature(),
        "test": lambda: CartoonsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
